apis/python/src/tiledbsoma/_dataframe.py

```diff
@@ -401,6 +401,30 @@ def get_enumeration_values(
 
         return self._handle.get_enumeration_values(column_names)
 
+    def extend_enumeration_values(
+        self: DataFrame, values: dict[str, pa.Array], dupes_ok: bool = False
+    ) -> None:
+        """Extend enumeration values for each column defined in `values`.
+        Raises ``ValueError`` if any of the the specified column names is not in
+        the schema, or if any is not of Arrow dictionary type. All the values
+        must provided must not already exist in the schema (see the output of
+        ``get_enumeration_values``), unless ``dupes_ok=True`` is provided."""
+
+        self._check_open_write()
+
+        # These assertions could be done in C++. However, it's easier here
+        # to do the exception-type multiplexing, raising ValueError for one
+        # thing, TileDBSOMAError for another.
+        for column_name in values.keys():
+            # As with get_enumeration_values: we are trusting pyarrow to raise
+            # KeyError, and raise it with a sufficiently clear error message,
+            # when the column name is not present within the schema.
+            field = self.schema.field(column_name)
+            if not isinstance(field.type, pa.DictionaryType):
```

Comment:
I believe the suggested way to do this with Arrow is:
> if not pa.types.is_dictionary(field.type):

Reply:
pushed